### PR TITLE
Remove unreachable condition

### PR DESCRIPTION
**Pull request by kravciak**

### CODE REVIEW

File: lib/caasp.pm [hunk 1]
```diff
@@ -153,9 +153,6 @@ sub process_reboot {
         elsif (match_has_tag 'grub2') {
             send_key 'ret';
         }
-        else {
-            die 'Kubic did not manage to reboot properly';
```
Comment (okurz) [inline lib/caasp.pm]: Why should this code be unreachable, isn't an explicit else branch always a good idea?

Comment (foursixnine) [reply lib/caasp.pm]: Wouldn't the assert_screen basically finish the test if the  needles with the corresponding task are not found?, therefore making that branch, indeed, as he's covering the only two possible scenarios?.

File: tests/caasp/transactional_update.pm [hunk 2]
```diff
@@ -81,7 +81,10 @@ sub run {
 
     # Check that zypper does not return 0 if update was aborted
     record_info 'Broken pkg', 'Install broken package poo#18644 - snapshot #3';
-    if (is_caasp('DVD')) {
+    if (is_caasp('=4.0')) {
+        record_info 'Test skipped - broken image needs breaking again';
```
Comment (foursixnine) [inline tests/caasp/transactional_update.pm]: is this a bug (ergo bsc?)? or expected behaviour?

File: lib/caasp.pm (hunk 1 above)
Comment (rwx788) [reply lib/caasp.pm]: In this matter I plan to create a ticket, that we can put the comment explaining check for all `assert_*` methods and not only `assert_script_run`. And then we can force it so no new assert is added without such message. It should explain review, as sometimes we do complex parcing and it's not clear what it is all about.

**Comment by foursixnine** [comment]
Merging as @kravciak is on vacations

File: tests/caasp/transactional_update.pm (hunk 2 above)
Comment (kravciak) [reply tests/caasp/transactional_update.pm]: thanks for merge :) - this is bug in tests, not product bug.
We have package that should fail during zypper install. Now it installs fine cause package specfile needs updating.

I will fix that when I am back from vacation, I just didn't want test to fail until then.

**Pull request merged**